pkg/service/auth_context.go

```diff
@@ -47,71 +48,122 @@ func NewAuthContext(parentCtx context.Context, req *auth.CheckRequest, apiConfig
 
 }
 
-func (authContext *AuthContext) evaluateAuthConfig(ctx context.Context, config common.AuthConfigEvaluator, cb configCallback) error {
+func (authContext *AuthContext) evaluateAuthConfig(ctx context.Context, config common.AuthConfigEvaluator, cb evaluateCallback) error {
 	select {
```

Comment:
What is the purpose of this select? 

It will not cancel the job if any other job returns error.